### PR TITLE
Spec reorganization pt 1

### PR DESCRIPTION
**Pull request by jdolitsky**
Related to #126

The following sections have been removed:
- Changes
- Historical Context (already in reorg)
- Overview / Listing Repositories (unused catalog API)

The following sections have been moved into commented reorg:
- Notational Conventions ---> Definitions / Document Language
- Overview / Errors ---> HTTP API / Error Codes
- Overview / API Version Check ---> Conformance / Workflow Categories / Pull
- Overview / Pulling An Image ---> Conformance / Workflow Categories / Pull
- Overview / Pushing An Image ---> Conformance / Workflow Categories / Push
- Overview / Listing Image Tags ---> Conformance / Workflow Categories / Content Discovery
- Overview / Deleting an Image ---> Conformance / Workflow Categories / Content Management
- Detail / Errors ---> HTTP API / Error Codes

### CODE REVIEW

**Comment by vbatts** [comment]
ohman. Heavy weight champ over here!

File: spec.md [hunk 1]
```diff
@@ -4134,591 +4046,641 @@ The error codes that MAY be included in the response body are enumerated below:
...
+- **Artifact**: a single piece of content, made up of a manifest and one or more layers
+- **Manifest**: a JSON document which defines an artifact
+- **Layer**: a single part of all the parts which comprise an artifact
+- **Config**: a special layer defined at the top of a manifest containing artifact metadata
+- **Blob**: a single binary content stored in a registry
+- **Digest**: a unique blob identifier
+- **Content**: a general term for content that can be downloaded from a registry (manifest or blob)
 
-###### On Failure: Not Found
+### Document Language
```
Comment (rchincha) [inline spec.md:4117]: Did you want to move this to the end of the document? RFCs typically put this language at the very beginning in a "Terminology" section.

**Comment by rchincha** [comment]
@jdolitsky, can we break this into a couple of baby steps.
Mainly edit-in-place as much as possible - renaming sections, expected error codes, etc. 
Will make it far easier to review.

**Comment by jdolitsky** [comment]
Closing this as un-mergeable/un-reviewable. Will come back will more PRs

**Pull request closed without merging**